lib/io.js

```diff
@@ -654,7 +654,7 @@
       disconnect: function() {
         logger.debugf('Disconnect transport');
         emitter.removeListener('router', onRouter);
-        return router.disconnect();
+        return (router ? router.disconnect() : null);
```

Comment:
Not sure router could be null in this case. Even if ping returns an error, if the topology has been updated, it's updated. Even on first connection, transport.connect() method should set a fixed router as soon as one of the connections succeds.

Reply:
This line change is a side-effect of the previous change (transport.disconnect() in infinispan.js). 
1. the transport.disconnect() in catch clause in infinispan.js is added as a solution for this bug, because when the client tries to connect to not-configured cache, it gives an error, the transport is not closed and the connection hangs.
2. on the other hand, with this solution some of the ssl  tests start to fail with message: 
Error: [Cannot read property 'disconnect' of undefined] is not any of: [CERT_SIGNATURE_FAILURE,self signed certificate]
and this error is thrown on router.disconnect() in io.js file. That's why I have added the null check.

Reply:
Oh I see, the SSL errors are client side errors that can happen on startup. In this case, I'd make the `disconnect()` function clearer of what it does. So, I'd check with `f.existy(router)` when `disconnect()` is called and the log accordingly. It can be confusing for the log message to say `Disconnect transport` when you actually don't do any disconnection.

Reply:
Also, always use `f.existy()` to check if a value is undefined or not. Also, better than returning `null`, return a Promise completed with empty, e.g. `Promise.resolve()`

Reply:
Thanks a lot Galder for your comments. I have fixed this part of code as you have suggested.